File: day4_part1_2019.py

with open('input.txt', 'r') as file:
    start, end = map(int, file.readline().strip().split('-'))

count = 0
for num in range(start, end+1):
    digits = [int(d) for d in str(num)]
    if sorted(digits) == digits and any(digits.count(d) >= 2 for d in set(digits)):
        count += 1

print(count)

File: day4_part1_2019.clj

(defn valid-password? [password]
  (let [digits (map #(Character/getNumericValue %) (str password))
        adjacent-same? (some #(apply = %) (partition 2 1 digits))
        increasing? (apply <= digits)]
    (and adjacent-same? increasing?)))

(defn count-valid-passwords [start end]
  (count (filter valid-password? (range start (inc end)))))

(def input (slurp "input.txt"))
(let [[start end] (map read-string (clojure.string/split input #"-"))]
  (println (count-valid-passwords start end)))
